ConfigurationSystem/Client/VOMS2CSSyncronizer.py

```diff
@@ -0,0 +1,472 @@
+""" VOMS2CSSyncronizer is a helper class containing the logic for synchronization
+    of the VOMS user data with the DIRAC Registry
+"""
+
+from collections import defaultdict
+
+from DIRAC.ConfigurationSystem.Client.CSAPI import CSAPI
+from DIRAC.Core.Security.VOMSService import VOMSService
+from DIRAC import S_OK, S_ERROR, gLogger, gConfig
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getVOOption, getVOMSRoleGroupMapping, \
+    getUsersInVO, getAllUsers
+from DIRAC.Core.Utilities.List import fromChar
+from DIRAC.Core.Utilities.PrettyPrint import printTable
+
+__RCSID__ = "$Id$"
+
+
+class VOMS2CSSynchronizer(object):
+
+  def __init__(self, vo, autoModifyUsers=True, autoAddUsers=True, autoDeleteUsers=False):
+
+    self.log = gLogger.getSubLogger("VOMS2CSSynchronizer")
+    self.csapi = CSAPI()
+    self.vo = vo
+    self.vomsVOName = getVOOption(vo, "VOMSName", "")
+    if not self.vomsVOName:
+      raise Exception("VOMS name not defined for VO %s" % vo)
+    self.adminMsgs = {'Errors': [], 'Info': []}
+    self.vomsUserDict = {}
+    self.autoModifyUsers = autoModifyUsers
+    self.autoAddUsers = autoAddUsers
+    self.autoDeleteUsers = autoDeleteUsers
+
+  def syncCSWithVOMS(self):
```

Comment:
This method is super long, and difficult to read. Can it be split in few functions? And then unit-tested.